File: 1. cpp/1. basics/lecture 2/p1.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
      for(int i=1;i<=n;i++) {
        for(int j=1;j<=n;j++) {
          cout<<"* ";
        }
        cout<<endl;
      }
    }
};

/* pattern output:

* * * *
* * * *
* * * *
* * * *

*/

int main() {
  int size;
  cout<<"Enter size of pattern:"<<endl;
  cin>>size;

  cout<<"directly constructor based output:"<<endl;
  solution pattern(size);

}

File: 4. kotlin/1. basics/lecture 2/p1.kt
import java.util.Scanner

class Solution(n: Int) {
    init {
        for (i in 1..n) {
            for (j in 1..n) {
                print("* ")
            }
            println()
        }
    }
}

/* pattern output:

* * * *
* * * *
* * * *
* * * *

*/

fun main() {
    val scanner = Scanner(System.`in`)
    println("Enter size of pattern:")
    val size = scanner.nextInt()

    println("directly constructor based output:")
    Solution(size)
}